### PR TITLE
Specify CENC

### LINKED ISSUE #820: Implement CENC as one of v1.1 features
A proposed location is 6.3 CENC before the "Codecs Parameter String" (aligned with that of av1-isobmff).

### PR DESCRIPTION
**Pull request by felicialim**
Fix #820

***
<a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/pull/827.html" title="Last updated on Jun 17, 2024, 11:25 PM UTC (406e96b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/827/ac94d8d...406e96b.html" title="Last updated on Jun 17, 2024, 11:25 PM UTC (406e96b)">Diff</a>

### CODE REVIEW

**Review by sunghee-hwang — APPROVED** [review]

**Comment by yilun-zhangs** [comment]
@felicialim 
Dose it need to specify which encryption types will be supported? `full sample encryption`, `subsample encryption`, 
`pattern encryption`, one or more.

File: index.bs [hunk 1]
```diff
@@ -253,6 +257,12 @@ url: https://www.iso.org/standard/77752.html#; spec: MP4-PCM; type: dfn;
 		"status": "Standard",
 		"publisher": "IETF",
 		"href": "https://tools.ietf.org/html/rfc8486"
+	},
+	"CENC": {
+		"title": "Information technology — MPEG systems technologies - Part 7: Common encryption in ISO base media file format files",
+		"status" : "Standard",
+		"publisher" : "ISO/IEC",
+		"href" : "https://www.iso.org/standard/84637.html"
```
Comment (sunghee-hwang) [inline index.bs:265]:
```suggestion
		"href" : "https://www.iso.org/standard/68042.html"
```

File: index.bs [hunk 2]
```diff
@@ -2058,6 +2068,10 @@ NOTE: Per the restriction of the profiles carried in an [=IA Track=], all [=Audi
 
 NOTE: In typical cases, when a track contains a single [=IA Sequence=], trimming can only happen at the beginning or the end of the [=IA Sequence=]. Therefore, the edit list can describe the start and end trimming with a single edit entry. A track storing consecutive [=IA Sequence=]s may need multiple edits in the edit list.
 
+## Common Encryption ## {#commonencryption}
+
+[=IA Track=]s MAY be protected. If protected, they SHALL conform to [[!CENC]]. Files SHOULD be protected using the [=cbcs=] protection scheme and MAY be protected using the [=cenc=] protection scheme.
```
Comment (sunghee-hwang) [inline index.bs:2073]:
```suggestion
[=IA Track=]s MAY be protected. If protected, they SHALL conform to [[!CENC]].
```

File: index.bs [hunk 3]
```diff
@@ -118,6 +118,10 @@ url: https://www.iso.org/standard/77752.html#; spec: MP4-PCM; type: dfn;
 	text: format_flags
 	text: PCM_sample_size
 	
+url: https://www.iso.org/standard/84637.html#; spec: CENC; type: dfn;
+	text: cenc
+	text: cbcs
+
```
Comment (sunghee-hwang) [inline index.bs:124]:
```suggestion
```

**Review by sunghee-hwang — CHANGES_REQUESTED** [review]
- The version of CENC reference is aligned with that of av1-isobmff.
- The restriction for encryption modes is removed. (to reflect WG's comment)

**Comment by felicialim** [comment]
> @felicialim Dose it need to specify which encryption types will be supported? `full sample encryption`, `subsample encryption`, `pattern encryption`, one or more.

According to the ISOBMFF spec, (whole-block) full sample should be used for audio. I've added some text to clarify it, please take a look.

**Review by sunghee-hwang — APPROVED** [review]
LGTM

File: index.bs [hunk 4]
```diff
@@ -1938,7 +1949,7 @@ NOTE: In a typical case, the OBUs in the first [=Descriptors=] of an [=IA Sequen
 A file conformant to this specification satisfies the following:
 - It SHALL conform to the normative requirements of [[!ISO-BMFF]].
 - It SHALL have the <dfn export for="ISO-BMFF Brand">iamf</dfn> brand among the compatible brands array of the FileTypeBox.
-- It SHALL contain at least one track using an [=IASampleEntry=].
+- It SHALL contain at least one track using an [=IASampleEntry=], possibly transformed by encryption as specified in [[#commonencryption]].
```
Comment (cconcolato) [inline index.bs:1952]: Strictly speaking an "old" implementation of v1.0 could have assumed that files would never have encrypted IAMF tracks. I wonder if this warrants a new brand to signal the possible encryption.

File: index.bs [hunk 5]
```diff
@@ -2058,6 +2069,12 @@ NOTE: Per the restriction of the profiles carried in an [=IA Track=], all [=Audi
 
 NOTE: In typical cases, when a track contains a single [=IA Sequence=], trimming can only happen at the beginning or the end of the [=IA Sequence=]. Therefore, the edit list can describe the start and end trimming with a single edit entry. A track storing consecutive [=IA Sequence=]s may need multiple edits in the edit list.
 
+## Common Encryption ## {#commonencryption}
+
+[=IA Track=]s MAY be protected. If protected, they SHALL conform to [[!CENC]].
+
+When the protection schemes [=cenc=] or [=cbc1=] are used, an [=IA Track=] SHALL be protected using full sample encryption. When the protection schemes [=cens=] or [=cbcs=] are used, an [=IA Track=] SHALL be protected using whole-block full sample encryption.
```
Comment (cconcolato) [inline index.bs:2076]: This is a lot of options, and therefore a lot of testing, and possible market fragmentation. Can't the group agree on one approach only?

File: index.bs (hunk 4 above)
Comment (tdaede) [reply index.bs:1952]: We already have box presence to signal CENC presence. We decided to not require new profiles to use CENC. If there are other examples of codecs that use brands to signal encryption presence we can do the same.

File: index.bs (hunk 5 above)
Comment (tdaede) [reply index.bs:2076]: The problem is CENC is mandatory in ISOBMFF, but CBCS is more widely used. We could pretty easily reduce it to two. Reducing it to one might be more difficult.

**Pull request merged**